changes/4311.housekeeping

```diff
@@ -0,0 +1 @@
+Updated CI pull request workflow for `next` branch to remove some redundant tests, run more tests in parallel and test the container build.
```

Comment:
ChangeLog test still failing for some reason, maybe we need to merge `next` into this branch?

Reply:
No, that change is still in develop. I think we're ok to merge with failing changelog as long as we plan on merging develop into next before the next 2.0 release.